NBitcoin/Protocol/Node.cs

```diff
@@ -528,46 +540,61 @@ public static Node Connect(Network network, NodeConnectionParameters parameters
 		public static Node Connect(Network network, NodeConnectionParameters parameters, Func<IPEndPoint[]> connectedEndpoints, Func<IPEndPoint, byte[]> getGroup = null)
 		{
 			getGroup = getGroup ?? new Func<IPEndPoint, byte[]>((a) => IpExtensions.GetGroup(a.Address));
-			connectedEndpoints = connectedEndpoints ?? new Func<IPEndPoint[]>(() => new IPEndPoint[0]);
+
+			if (connectedEndpoints() == null)

```

Comment:
I think this behaviours is different `connectedEndpoints()` you are executing the func instead of checking is its null.

Reply:
The only code that calls this method is:

```
public static Node Connect(Network network, NodeConnectionParameters parameters = null, IPEndPoint[] connectedEndpoints = null, Func<IPEndPoint, byte[]> getGroup = null)
{
 return Connect(network, parameters, new Func<IPEndPoint[]>(() => connectedEndpoints), getGroup);
}
```

which wraps the connectedEndpoints in a method, of which the result of can be null.

I had to make this change as some unit tests were failing because of this logical error.

i.e. connectedEndpoints will never be null with the old code:

```connectedEndpoints = connectedEndpoints ?? new Func<IPEndPoint[]>(() => new IPEndPoint[0]); ```

Reply:
Ah I think we might also be calling it from the fullnode, double check that.

Reply:
OK I checked and fullnode does not call it either :)